### PR TITLE
Companies show authorization Closes #64

### PR DESCRIPTION
**Pull request by oorion**
Add authorization for companies show page.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -16,4 +16,10 @@ def authorize!
       redirect_to root_path
     end
   end
+
+  def authorize_company!
+    unless (current_user && (current_user.admin? || current_user.company.id.to_s == params[:id]))
+      redirect_to root_path
+    end
+  end
```
Comment (KristaANelson) [inline app/controllers/application_controller.rb:24]: :+1:

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -16,4 +16,10 @@ def authorize!
       redirect_to root_path
     end
   end
+
+  def authorize_company!
+    unless (current_user && (current_user.admin? || current_user.company.id.to_s == params[:id]))
+      redirect_to root_path
+    end
```
Comment (scottalexandra) [inline app/controllers/application_controller.rb:23]: Great! I like that it checks that company you are associated with and compares that to the company page you are trying to visit.

File: spec/features/company_authorization_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    fill_in 'user[email]', with: company.users.first.email
+    fill_in 'user[password]', with: 'password'
+    click_link_or_button('Log in')
+    visit company_path(other_company.id)
+
+    expect(current_path).to eq(root_path)
+    expect(page).to_not have_content(company.organization)
+  end
+end
```
Comment (KristaANelson) [inline spec/features/company_authorization_spec.rb:29]: :tada: nice tests!

Comment (scottalexandra) [inline spec/features/company_authorization_spec.rb:29]: Looks good! :space_invader: :sandal: :sake:

**Comment by KristaANelson** [comment]
:sparkles:  Nice PR!! I really liked that it wan't too big! Nice tests! :sparkles:

**Pull request merged**